include/nbl/video/CThreadSafeGPUQueueAdapter.h

```diff
@@ -14,8 +14,8 @@ namespace nbl::video
         core::smart_refctd_ptr<IGPUQueue> originalQueue = nullptr;
         std::mutex m;
     public:
-        CThreadSafeGPUQueueAdapter(nbl::core::smart_refctd_ptr<IGPUQueue>&& originalQueue) : IGPUQueue(nullptr, originalQueue->getFamilyIndex(), originalQueue->getFlags(), originalQueue->getPriority()),
-            originalQueue(originalQueue) {}        
+        CThreadSafeGPUQueueAdapter(nbl::core::smart_refctd_ptr<IGPUQueue>&& original) : IGPUQueue(nullptr, original->getFamilyIndex(), original->getFlags(), original->getPriority()),
+            originalQueue(original) {}        
```

Comment:
still need to do `originalQueue(std::move(original))`

without an `std::move` an r-value reference will always decay to an l-value reference when passed further along